pkg/istio/rbac/v1/provider.go

```diff
@@ -96,15 +101,167 @@ func (p *v1) ConvertAthenzModelIntoIstioRbac(m athenz.Model) []model.Config {
 // GetCurrentIstioRbac returns the ServiceRole and ServiceRoleBinding resources for the specified model's namespace
 func (p *v1) GetCurrentIstioRbac(m athenz.Model, csc model.ConfigStoreCache) []model.Config {
 
-	sr, err := csc.List(model.ServiceRole.Type, m.Namespace)
+	sr, err := csc.List(collections.IstioRbacV1Alpha1Serviceroles.Resource().GroupVersionKind(), m.Namespace)
 	if err != nil {
 		log.Errorf("Error listing the ServiceRole resources in the namespace: %s", m.Namespace)
 	}
 
-	srb, err := csc.List(model.ServiceRoleBinding.Type, m.Namespace)
+	srb, err := csc.List(collections.IstioRbacV1Alpha1Servicerolebindings.Resource().GroupVersionKind(), m.Namespace)
 	if err != nil {
 		log.Errorf("Error listing the ServiceRoleBinding resources in the namespace: %s", m.Namespace)
 	}
 
 	return append(sr, srb...)
 }
+
+// ConvertAthenzModelIntoIstioAuthzPolicy converts the Athenz RBAC model into Istio Authorization V1Beta1 specific
+// RBAC custom resource (AuthorizationPolicy)
+func (p *v1) ConvertAthenzModelIntoIstioAuthzPolicy(athenzModel athenz.Model, namespace string, serviceName string, svcLabel string) model.Config {
+	// authz policy is created per service. each rule is created by each role, and form the rules under
+	// this authz policy.
+	var out model.Config
+	// form authorization config meta
+	// namespace: service's namespace
+	// name: service's name
+	schema := collections.IstioSecurityV1Beta1Authorizationpolicies
+	out.ConfigMeta = model.ConfigMeta{
+		Type:      schema.Resource().Kind(),
+		Group:     schema.Resource().Group(),
+		Version:   schema.Resource().Version(),
+		Namespace: namespace,
```

Comment:
We could potentially skip passing the namespace as an argument as we have the athenz domain name in the athenz model object.